### DISCUSSION TITLE
Supported Library , Datatypes , Querytype and Index types

### DISCUSSION THREAD

**Discussion by pauldintel** [milvus-io/milvus#28027] [Q&A and General discussion]
hi Team , 
Like other vector DB I am looking what Libraries and Index type from below is supported by Milvus 

HNSWlib
DiskANN / Vamana
ScaNN
FAISS-IVFP
NGT

Also are all these datatypes supported ?
FLOAT
INT8 , UINT8
FLOAT16
BFLOAT16

And does Milvus has support of Intel AVX instructions like AVX512 ?

What are the query types from below it supports ?

Search
Filter + Search

thanks

**Comment by yhmo** [comment]
HNSWlib ------------------------------------------- supported
DiskANN / Vamana------------------------------- supported
ScaNN----------------------------------------------- supported
FAISS-IVFP---------------------------------------- supported, I suppose it is IVF_FLAT
NGT-------------------------------------------------- not supported

FLOAT----------------------------------------------- supported
INT8 ------------------------------------------------- supported
UINT8------------------------------------------------ not supported
FLOAT16-------------------------------------------- not supported
BFLOAT16------------------------------------------ not supported

AVX512 --------------------------------------------- supported


Search---------------------------------------------- supported
Filter + Search------------------------------------ supported

**Comment by pauldintel** [comment]
thank you so much @yhmo - appreciate the details ..
